src/plugins/controls/public/range_slider/embeddable/range_slider_embeddable.tsx

```diff
@@ -169,19 +173,21 @@ export class RangeSliderEmbeddable extends Embeddable<RangeSliderEmbeddableInput
 
     // fetch available min/max when input changes
     this.subscriptions.add(
-      dataFetchPipe.subscribe(async () =>
-        this.runRangeSliderQuery().catch((e) => {
+      dataFetchPipe.subscribe(async () => {
+        try {
+          await this.runRangeSliderQuery();
+          await this.buildFilter();
+        } catch (e) {
           this.dispatch.setErrorMessage(e.message);
-        })
-      )
+        }
+      })
     );
 
-    // build filters when value change
+    // build filters when value changes
     this.subscriptions.add(
       this.getInput$()
         .pipe(
-          debounceTime(400),
-          distinctUntilChanged((a, b) => isEqual(a.value, b.value)),
+          distinctUntilChanged((a, b) => isEqual(a.value ?? ['', ''], b.value ?? ['', ''])),
```

Comment:
I moved the debounce to the component itself - this limits the amount of input changes that are fired, which makes the dragging feel a lot smoother (although it's still not perfect).